packages/ses/src/make-hardener.js

```diff
@@ -225,23 +260,65 @@ export const makeHardener = () => {
         setForEach(toFreeze, freezeAndTraverse);
       }
 
-      /** @param {any} value */
-      function markHardened(value) {
-        weaksetAdd(hardened, value);
+      function checkProtos() {
+        if (intrinsicsWereHardened) {
+          if (setGetSize(protosToCheck) > 0) {
+            const unhardenedProtoPaths = [];
+            setForEach(protosToCheck, proto => {
+              arrayPush(unhardenedProtoPaths, weakmapGet(paths, proto));
+            });
+            throw TypeError(
+              `Expected the following paths to be hardened: ${arrayJoin(
+                unhardenedProtoPaths,
+                ', ',
+              )}`,
+            );
+          }
+          if (hardenedAtLockdownSize > 0) {
+            const actuallyHardenedIntrinsics = new Set();
+
+            setForEach(toFreeze, value => {
+              if (weaksetHas(hardenedAtLockdown, value)) {
+                setAdd(actuallyHardenedIntrinsics, value);
+              }
+            });
+            const actuallyHardenedIntrinsicsSize = setGetSize(
+              actuallyHardenedIntrinsics,
+            );
+            if (actuallyHardenedIntrinsicsSize !== hardenedAtLockdownSize) {
+              throw TypeError(
+                `Not all expected prototypes were hardened (expected=${hardenedAtLockdownSize}, got=${actuallyHardenedIntrinsicsSize})`,
+              );
+            } else {
+              hardenedAtLockdownSize = 0;
+              setForEach(actuallyHardenedIntrinsics, value => {
+                setDelete(hardenedAtLockdown, value);
+              });
+            }
+          }
+        } else if (setGetSize(protosToCheck) > 0) {
```

Comment:
If intrinsics were note hardened, we should fail for any objects we intend to harden, but that were already in the `hardenedAtLockdown` set. (enforcing the rule that a prototype must be hardened before its instances)